linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -6296,13 +6247,10 @@ private[optimizer] object OptimizerCore {
   private object PreTransTree {
     def apply(tree: Tree): PreTransTree = {
       val refinedTpe: RefinedType = BlockOrAlone.last(tree) match {
-        case _:LoadModule | _:NewArray | _:ArrayValue | _:ClassOf =>
-          RefinedType(tree.tpe, isExact = true, isNullable = false)
-        case GetClass(x) if x.tpe != AnyType && x.tpe != ClassType(ObjectClass) =>
-          /* If x.tpe is neither AnyType nor j.l.Object, it cannot be a JS
-           * object, so its getClass() cannot be null.
-           */
-          RefinedType(tree.tpe, isExact = true, isNullable = false)
+        case _:New | _:NewArray | _:ArrayValue | _:ClassOf =>
+          RefinedType(tree.tpe, isExact = true)
+        case Transient(Cast(LoadModule(_), ClassType(_, false))) =>
```

Comment:
`MaybeCast`? (IIUC the cast type guard is unnecessary)

Reply:
Hum, that's not what I interpreted `isExact` to mean. So does this mean `nullable ==> !exact`? If so, probably we should have an assertion in `RefinedType`?

Reply:
> So does this mean `nullable ==> !exact`?

No: `NullType` is `nullable && exact`. ;)

Reply:
What matters the most about `exact` is that `if !isSubtype(tpe, T) && tpe.isExact`, then we know that `tpe` is disjoint from `T`.